pandas/core/indexes/period.py

```diff
@@ -799,6 +803,30 @@ def _wrap_setop_result(self, other, result):
         result.name = name
         return result
 
+    def intersection(self, other, sort=False):
+        self._validate_sort_keyword(sort)
+        self._assert_can_do_setop(other)
+        res_name = get_op_result_name(self, other)
+        other = ensure_index(other)
+
```

Comment:
can this code be shared with superclasses?

Reply:
i think so, its going to be a few steps before we get there